contexts/AuthProvider.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  ReactNode,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import axios from "@/lib/axios";
+import { LoginValues } from "@/types/login";
+import { useRouter } from "next/router";
+import { setCookie } from "cookies-next";
+import { User } from "@/types/user";
+
+interface AuthContextType {
+  user: any;
+  login: (values: LoginValues) => Promise<void>;
+  isPending: boolean;
+}
+
+const AuthContext = createContext<AuthContextType>({
+  user: null,
+  login: async () => {},
+  isPending: true,
+});
+
+interface AuthProviderProps {
+  children: ReactNode;
+}
+
+export function AuthProvider({ children }: AuthProviderProps) {
+  const [values, setValues] = useState<{
+    user: User | null;
+    isPending: boolean;
+  }>({
+    user: null,
+    isPending: true,
+  });
```

Comment:
@ssseeo0 

로딩 상태와 유저 정보는 같은 수준의 정보가 아니잖아요, 그래서 독립된 도메인으로 존재해야 할 것 같아요.
굳이 분리를 해서 보관하면 어떨까요?

```suggestion
const [user, setUser] = useState<User | null>(null)
const [isLoading, setIsLoading] = useState(true)
```